### ISSUE TITLE
confidence map

### ISSUE THREAD

**Issue by LeiyuanMa** [hfslyc/AdvSemiSeg#24]
Excuse me, There are one more questions...hope it not bothing you too much...
"The output of D is a probability map with values ranges from 0-1. When the input is unlabeled data, we use the D output map as the indicator for semi-supervised learning."

D_out = interp(model_D(F.softmax(pred)))
D_out_sigmoid = F.sigmoid(D_out).data.cpu().numpy().squeeze(axis=1)
        ##following is used to draw the confindence map in your papar Figure3
        for i in range(args.batch_size):
               plt.imshow(D_out_sigmoid[i], cmap='jet')
               plt.show()
Can I interpret the D_out_sigmoid is the confidence map you mentioned?
I use VOC_20000.pth and VOC_20000_D.pth(when use these two model the evaluatation mIOU is 0.709) as the parameters for RESTORE_FROM and RESTORE_FROM_D,and tried to get the figure of confidence map,but it looks not matching so well,it looks weird ,can you explain how you get the confidence map in your paper?

**Comment by hfslyc** [comment]
the code looks fine to me. What do you mean by "not matching so well"?

**Comment by LeiyuanMa** [comment]
plt.imshow(D_out_sigmoid[i], cmap='jet')
Does this image should looks close to the ground truth?

**Comment by LeiyuanMa** [comment]
and Is it the right way to draw confidence map?

**Comment by hfslyc** [comment]
the map is not related to ground truth. It's the discriminator score on how
close it is to the ground truth.
